enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/presencecheck/iproov/provider/IProovPresenceCheckProvider.java

```diff
@@ -77,31 +79,26 @@ public IProovPresenceCheckProvider(
     }
 
     @Override
-    public void initPresenceCheck(OwnerId id, Image photo) throws PresenceCheckException, RemoteCommunicationException {
-        ResponseEntity<String> responseEntityToken = callGenerateEnrolToken(id);
-        // FIXME temporary solution of repeated presence check initialization
-        // Deleting the iProov enrollment properly is not implemented yet, use random suffix to the current userId
-        if (responseEntityToken.getBody() != null && responseEntityToken.getBody().contains("already_enrolled")) {
-            logger.warn("Retrying the iProov enrollment with adapted userId");
-            final OwnerId adaptedId = new OwnerId();
-            adaptedId.setUserId(id.getUserId() + Calendar.getInstance().toInstant().getEpochSecond());
-            adaptedId.setActivationId(id.getActivationId());
-            adaptedId.setTimestamp(id.getTimestamp());
-            id = adaptedId;
-            responseEntityToken = callGenerateEnrolToken(id);
-        }
+    public void initPresenceCheck(final OwnerId id, final Image photo) throws PresenceCheckException, RemoteCommunicationException {
+        final ResponseEntity<String> responseEntityToken = callGenerateEnrolToken(id);
 
-        if (responseEntityToken.getBody() == null) {
+        final String body = responseEntityToken.getBody();
+        if (body == null) {
             throw new RemoteCommunicationException("Missing response body when generating an enrol token in iProov, " + id);
         }
 
+        // Deleting the iProov enrollment properly is not implemented yet
+        if (body.contains("already_enrolled")) {
```

Comment:
Simply raising the exception will deadlock the process. 
We need to:
- If the photo was enrolled in this process, we could move to the verification cycle. (if photo enroll fails on our side to properly update the process)
- If the photo was enrolled in this process, but the whole cycle is repeated (i.e. new documents)
- If this is a new process, we need to get the enrolled token from previous attempt (or use alternated userId)

There could be controls on other parts of the process to ensure this will not happened, but I thing currently there are not.

Reply:
This should never happen, just a handbrake. The real check is in the layer above https://github.com/wultra/enrollment-server/pull/770/files#diff-ee0512b663943b0bbb18ad44c1551c55261a422493b7736049cd0ada225027f7R230

Reply:
🤔 But yes, I have to double-check `userId`

Reply:
- [ ] Improve handling `already_enrolled` based on the answer from iProov support.